internal/commands/relationship.go

```diff
@@ -124,57 +127,62 @@ func bulkDeleteRelationships(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	request := &v1.ReadRelationshipsRequest{RelationshipFilter: filter}
-
-	counter := -1
-	if cobrautil.MustGetBool(cmd, "estimate-count") {
-		request.Consistency = &v1.Consistency{Requirement: &v1.Consistency_FullyConsistent{FullyConsistent: true}}
-
-		ctx, cancel := context.WithCancel(cmd.Context())
-		defer cancel()
+	bar := console.CreateProgressBar("deleting relationships")
+	defer func() {
+		_ = bar.Finish()
+	}()
 
-		log.Trace().Interface("request", request).Send()
-		resp, err := spicedbClient.ReadRelationships(ctx, request)
+	allowPartialDeletions := cobrautil.MustGetBool(cmd, "force")
+	optionalLimit := cobrautil.MustGetUint(cmd, "optional-limit")
+	var resp *v1.DeleteRelationshipsResponse
+	for {
+		delRequest := &v1.DeleteRelationshipsRequest{
+			RelationshipFilter:            filter,
+			OptionalLimit:                 uint32(optionalLimit),
+			OptionalAllowPartialDeletions: allowPartialDeletions,
+		}
+		log.Trace().Interface("request", delRequest).Msg("deleting relationships")
+
+		resp, err = spicedbClient.DeleteRelationships(cmd.Context(), delRequest)
+		if errorInfo, ok := grpcErrorInfoFrom(err); ok {
+			if errorInfo.GetReason() == v1.ErrorReason_ERROR_REASON_TOO_MANY_RELATIONSHIPS_FOR_TRANSACTIONAL_DELETE.String() {
+				return fmt.Errorf("partial deletion of type %s prevented, consider increasing --optional-limit beyond %s or deleting all relationships using --force",
```

Comment:
I'd say something like: "could not delete %s, as found more than %d relationships"